src/main/java/org/mariadb/jdbc/internal/stream/PacketOutputStream.java

```diff
@@ -255,66 +225,75 @@ private void checkPacketMaxSize(int limit) throws MaxAllowedPacketException {
     }
 
     private void internalFlush() throws IOException {
-        buffer.flip();
-        int limit = buffer.limit();
-        if (limit > 0) {
-            checkPacketMaxSize(limit);
-
-            byte[] bufferBytes = new byte[0];
-            int notCompressPosition = 0;
-            int expectedPacketSize = limit + HEADER_LENGTH * ((limit / maxPacketSize) + 1);
-            if (useCompression) {
-                bufferBytes = new byte[expectedPacketSize];
-            }
-
-            while (notCompressPosition < expectedPacketSize) {
-                int length = buffer.remaining();
-                if (length > maxPacketSize) {
-                    length = maxPacketSize;
-                }
-                if (useCompression) {
-                    bufferBytes[notCompressPosition++] = (byte) (length & 0xff);
-                    bufferBytes[notCompressPosition++] = (byte) (length >>> 8);
-                    bufferBytes[notCompressPosition++] = (byte) (length >>> 16);
-                    bufferBytes[notCompressPosition++] = (byte) seqNo++;
-                } else {
-                    byte[] header = new byte[HEADER_LENGTH];
-                    header[0] = (byte) (length & 0xff);
-                    header[1] = (byte) (length >>> 8);
-                    header[2] = (byte) (length >>> 16);
-                    header[3] = (byte) seqNo++;
-                    outputStream.write(header);
-                    notCompressPosition += 4;
-                }
-
-                if (length > 0) {
-                    if (useCompression) {
-                        buffer.get(bufferBytes, notCompressPosition, length);
-                        notCompressPosition += length;
-                    } else {
-                        if (buffer.hasArray()) {
-                            outputStream.write(buffer.array(), buffer.position(), length);
-                            buffer.position(buffer.position() + length);
-                        } else {
-                            bufferBytes = new byte[length];
-                            buffer.get(bufferBytes, 0, length);
-                            outputStream.write(bufferBytes, 0, length);
-                        }
-                        notCompressPosition += length;
-                        outputStream.flush();
-                    }
-                }
-            }
-
-
-            if (useCompression) {
-                //now bufferBytes in filled with uncompressed data
-                compressedAndSend(notCompressPosition, bufferBytes);
-            }
-        }
-
-    }
-
+    	buffer.prepare();
```

Comment:
Okay, but where is the eclipse format for the established style ?

On Fri, Mar 4, 2016 at 3:24 PM, Mike Jensen <notifications@github.com>
wrote:

> In src/main/java/org/mariadb/jdbc/internal/stream/PacketOutputStream.java
> <https://github.com/MariaDB/mariadb-connector-j/pull/72#discussion_r55034881>
> :
>
> > -                        notCompressPosition += length;
> > -                        outputStream.flush();
> > -                    }
> > -                }
> > -            }
> > -
> > -
> > -            if (useCompression) {
> > -                //now bufferBytes in filled with uncompressed data
> > -                compressedAndSend(notCompressPosition, bufferBytes);
> > -            }
> > -        }
> > -
> > -    }
> > -
> > +    	buffer.prepare();
>
> I will let others comment, but these spacing changes really bother me.
> Mostly because it's completely inconsistent with the rest of the project.
> Is there a reason you don't want to develop in the already established
> style?
>
> If it does not bother @rusher <https://github.com/rusher> though, I am
> fine with trying to ignore it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MariaDB/mariadb-connector-j/pull/72/files#r55034881>.
>